File: backend-spring/src/main/java/com/codequest/repository/UserLearningProfileRepository.java
package com.codequest.repository;

import com.codequest.domain.entity.UserLearningProfile;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import java.util.Optional;

@Repository
public interface UserLearningProfileRepository extends JpaRepository<UserLearningProfile, Long> {
    
    Optional<UserLearningProfile> findByUserIdAndLanguageAndTrackAndLevelId(
            String userId, String language, String track, Integer levelId);
}

File: backend-spring/src/main/kotlin/com/codequest/repository/UserLearningProfileRepository.kt
package com.codequest.repository

import com.codequest.domain.entity.UserLearningProfile
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface UserLearningProfileRepository : JpaRepository<UserLearningProfile, Long> {
    
    fun findByUserIdAndLanguageAndTrackAndLevelId(
        userId: String,
        language: String,
        track: String,
        levelId: Int
    ): UserLearningProfile?
}

